products/iam/api.yaml

```diff
@@ -195,4 +198,236 @@ objects:
       - !ruby/object:Api::Type::Boolean
         name: 'deleted'
         description: The current deleted state of the role
-        output: true
\ No newline at end of file
+        output: true
+  - !ruby/object:Api::Resource
+    name: 'WorkloadIdentityPool'
+    min_version: beta
+    base_url: projects/{{project}}/locations/global/workloadIdentityPools
+    create_url: projects/{{project}}/locations/global/workloadIdentityPools?workloadIdentityPoolId={{id}}
+    description: |
+      Represents a collection of external workload identities. You can define IAM policies to
+      grant these identities access to Google Cloud resources.
+    update_mask: true
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: True
+        allowed:
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: |
+          The state of the pool.
+          STATE_UNSPECIFIED: State unspecified.
+          ACTIVE: The pool is active, and may be used in Google Cloud policies.
+          DELETED: The pool is soft-deleted. Soft-deleted pools are permanently deleted after
+          approximately 30 days. You can restore a soft-deleted pool using
+          UndeleteWorkloadIdentityPool. You cannot reuse the ID of a soft-deleted pool until it is
+          permanently deleted. While a pool is deleted, you cannot use it to exchange tokens, or
+          use existing tokens to access resources. If the pool is undeleted, existing tokens grant
+          access again.
+        output: true
+        values:
+          - :STATE_UNSPECIFIED
+          - :ACTIVE
+          - :DELETED
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: A display name for the pool. Cannot exceed 32 characters.
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: A description of the pool. Cannot exceed 256 characters.
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: The resource name of the pool.
+        output: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'disabled'
+        description: |
+          Whether the pool is disabled. You cannot use a disabled pool to exchange tokens, or use
+          existing tokens to access resources. If the pool is re-enabled, existing tokens grant
+          access again.
+  - !ruby/object:Api::Resource
+    name: 'WorkloadIdentityPoolProvider'
```

Comment:
Would you mind breaking this resource into a followup PR afterwards? In my experience it tends to be a lot easier if we're iterating on one resource at a time, especially because a lot of comments that apply to one resource will apply to the second.

Reply:
I've removed WorkloadIdentityPoolProvider for now and we can add that in a subsequent PR